src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -110,6 +113,155 @@ def validate(self, node1, node2, options=None):
                         f"{node1.view()}\n"
                         f"{node2.view()}"))
 
+        vars1 = VariablesAccessInfo(node1)
+        vars2 = VariablesAccessInfo(node2)
+
+        # Check if the loops have the same loop variable
+        loop_var1 = node1.variable
+        loop_var2 = node2.variable
```

Comment:
Could you try indenting the part that you added here inside the previous "else:", if the "if isinstance(node1, PSyLoop) and isinstance(node2, PSyLoop):" condition is taken we are in PSyKAL domain and the fact that there are no dependencies is guaranteed by the definition and not code analysis.

Reply:
Done, didn't break anything so probably its fine.